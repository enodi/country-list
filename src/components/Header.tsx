import React from "react";
import { NavLink } from "react-router-dom";

const Header = () => (
    <header className="header">
        <div className="header__content">
            <NavLink className="header__title" to="/">
                <h1>Countries List</h1>
            </NavLink>
        </div>
    </header>
);

export default Header;
