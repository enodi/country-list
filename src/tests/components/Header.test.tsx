import React from "react";
import { shallow } from "enzyme";

import Header from "../../components/Header";

describe("Header Component", () => {
    let wrapper: any;
    beforeEach(() => {
        wrapper = shallow(<Header />);
    });

    it("should render Header component correctly", () => {
        expect(wrapper).toMatchSnapshot();
    });

    it("should contain h1 element with text", () => {
        expect(wrapper.find("h1").text()).toEqual("Countries List");
    });

    it("should contain attribute that navigates to home page", () => {
        expect(wrapper.find("NavLink").prop("to")).toEqual("/");
    });
});
