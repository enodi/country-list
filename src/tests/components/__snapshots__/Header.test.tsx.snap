// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Component should render Header component correctly 1`] = `
<header
  className="header"
>
  <div
    className="header__content"
  >
    <NavLink
      className="header__title"
      to="/"
    >
      <h1>
        Countries List
      </h1>
    </NavLink>
  </div>
</header>
`;
